emcc.py

```diff
@@ -3582,6 +3583,35 @@ def preprocess_wasm2js_script():
     write_file(final_js, js)
 
 
+def node_es6_imports():
+  if not settings.EXPORT_ES6 or not shared.target_environment_may_be('node'):
+    return ''
+
+  if settings.USE_ES6_IMPORT_META:
+    import_script_url = 'import.meta.url'
+  else:
+    import_script_url = "pathToFileURL('.')"
+
+  # Avoid Node specific imports in browser.
+  # TODO: Swap all `require()`'s with `import()`'s?
+  if shared.target_environment_may_be('web'):
+    return '''
+if (typeof process == 'object' &&
+    typeof process.versions == 'object' &&
+    typeof process.versions.node == 'string') {
+  const { createRequire } = await import('module');
+  var { pathToFileURL, fileURLToPath } = await import('url');
+  var require = createRequire(%s);
+}
+''' % import_script_url
+  else:
+    return '''
+import { createRequire } from 'module';
+import { pathToFileURL, fileURLToPath } from 'url';
```

Comment:
Sorry I think this might have been an old comment that I hadn't send.. maybe can ignore this?

Reply:
No worries. The issue was that this modularize wrapper and imports are appended after Closure in ran, so it's not unaware of this. For the same reason, it currently requires the `createRequire` definition in `closure-externs.js` to avoid minification, if any.
https://github.com/emscripten-core/emscripten/blob/5ec36415577b9f8492c843134dc6e38ec48f4438/src/closure-externs/closure-externs.js#L18-L19

I avoided the use of the `pathToFileURL` and `fileURLToPath` imports here, since it's now only used by:
https://github.com/emscripten-core/emscripten/blob/5ec36415577b9f8492c843134dc6e38ec48f4438/src/shell.js#L213

For reference, here are the new static import declarations:
https://github.com/emscripten-core/emscripten/blob/5ec36415577b9f8492c843134dc6e38ec48f4438/emcc.py#L3702-L3706

Reply:
... I also tried to move these static imports to `pre_modularize` in commit 234940f032e4353a4390af02bbc59a6a9f87afcf locally, but that seems to cause a `java.lang.NullPointerException` when running `other.test_emcc_output_mjs_closure_node`.
<details>
  <summary>Details</summary>

```
building:ERROR: Closure compiler run failed:

building:ERROR: java.lang.NullPointerException: NAME createRequire$$module$module 3:16  [length: 13] [source_file: /tmp/tmpvxnkv4rn/emscripten_temp_012it4fl/hello_world.js.pre-modular.js] [original_name: createRequire] : Color{id=0, debugInfo=DebugInfo{compositeTypename=unknown}, prototypes=[], instanceColors=[], invalidating=true, propertiesKeepOriginalName=false, constructor=false, ownProperties=[], boxId=null, closureAssert=false, unionElements=[]}
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:907)
	at com.google.javascript.jscomp.RemoveUnusedCode.getVarForNameNode(RemoveUnusedCode.java:841)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseNameNode(RemoveUnusedCode.java:694)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseNode(RemoveUnusedCode.java:520)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseChildren(RemoveUnusedCode.java:1245)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseCall(RemoveUnusedCode.java:752)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseNode(RemoveUnusedCode.java:448)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseDeclarationStatement(RemoveUnusedCode.java:1026)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseNode(RemoveUnusedCode.java:500)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseChildren(RemoveUnusedCode.java:1245)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseNode(RemoveUnusedCode.java:530)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseChildren(RemoveUnusedCode.java:1245)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseNode(RemoveUnusedCode.java:530)
	at com.google.javascript.jscomp.RemoveUnusedCode.access$1200(RemoveUnusedCode.java:82)
	at com.google.javascript.jscomp.RemoveUnusedCode$Continuation.apply(RemoveUnusedCode.java:1762)
	at com.google.javascript.jscomp.RemoveUnusedCode.traverseAndRemoveUnusedReferences(RemoveUnusedCode.java:363)
	at com.google.javascript.jscomp.RemoveUnusedCode.process(RemoveUnusedCode.java:338)
	at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:267)
	at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:186)
	at com.google.javascript.jscomp.Compiler.performTranspilationAndOptimizations(Compiler.java:2572)
	at com.google.javascript.jscomp.Compiler.lambda$stage2Passes$5(Compiler.java:936)
	at com.google.javascript.jscomp.CompilerExecutor.lambda$runInCompilerThread$0(CompilerExecutor.java:101)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:597)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:194)
```
</details>